compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2280,7 +2280,16 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     pub fn get_attr(self, did: DefId, attr: Symbol) -> Option<&'tcx ast::Attribute> {
-        self.get_attrs(did, attr).next()
+        if cfg!(debug_assertions) {
+            let mut attrs = self.get_attrs(did, attr);
+            let first = attrs.next();
+            if attrs.next().is_some() {
+                bug!("get_attr: multiple attributes with same attr symbol");
+            }
+            first
```

Comment:
we can't check whether there are multiple attributes, but instead have to check whether we only use this function for cases where the compiler can reasonably deal with multiple attributes.

e.g. the following would ICE with your change (please add that as a test)
```rust
#[repr(C)]
#[repr(C)]
enum Foo {}

fn main() {}
```

instead check - similar to `is_builtin_only_local` - whether the feature definition in `rustc_feature`  either warns or errors on duplicate attributes. This might not be enough as there are some attributes where duplicates are allowed, but useless, but that's a bridge we can burn when we get to it.

Reply:
got it, I will dig more and ping you if I need help.

Reply:
We should check:

```rust
pub enum AttributeDuplicates {
    #[default]
    DuplicatesOk
    ...
}
```

https://github.com/rust-lang/rust/blob/8556e6620e4866526b3cea767ad8c20ae877a569/compiler/rustc_feature/src/builtin_attrs.rs#L101

If the attribute is marked as `DuplicatesOk` `WarnFollowing`, or `WarnFollowingWordOnly`, then we allow multiple attributes, otherwise raise an error?

Reply:
We have a check for proper diagnostics,
https://github.com/rust-lang/rust/blob/e1b28cd2f16bd5b832183d7968cae3bb9213e78d/compiler/rustc_passes/src/check_attr.rs#L2223

Reply:
> If the attribute is marked as `DuplicatesOk` `WarnFollowing`, or `WarnFollowingWordOnly`, then we allow multiple attributes, otherwise raise an error?

The other way around. We should only use `get_attr` for attributes which are either `WarnFollowing`, `ErrorFollowing`, `ErrorPreceding`, `FutureWarnFollowing` and `FutureWarnPreceding`. For all other attributes using `get_attr` is a **compiler** bug so we should ICE there. We should not emit errors for users in `get_attr`.

Reply:
@lcnr Code updated and added a testcase.